devito/types/sparse.py

```diff
@@ -669,8 +675,11 @@ def interpolate(self, expr, offset=0, increment=False, self_subs={}):
 
         variables = list(retrieve_function_carriers(expr))
 
+        # Need to get origin of the field in case it is staggered
+        field_offset = variables[0].origin
```

Comment:
why is `0` special here?

Reply:
nothing special, just take the first, I need to add a TODO as if multiple variable need to do for each different staggering pearately